docs: add user guide (#1292)
docarray/docarray: docs/user_guide/intro.md

# User Guide - Introduction

This user guide shows you how to use `DocArray` with most of its features.

There are three main sections:

- [Representing Data](representing/first_step.md): This section will show you how to use `DocArray` to represent your data. This is a great starting point if you want to better organize the data in your ML models, or if you are looking for a "pydantic for ML".
- [Sending Data](sending/first_step.md): This section will show you how to use `DocArray` to send your data. This is a great starting point if you want to serve your ML model, for example through FastAPI.
- [Storing Data](storing/first_step.md): This section will show you how to use `DocArray` to store your data. This is a great starting point if you are looking for an "ORM for vector databases".

You should start by reading the [Representing Data](representing/first_step.md) section, and then the [Sending Data](sending/first_step.md) and [Storing Data](storing/first_step.md) sections can be read in any order.

You will first need to install `DocArray` in your Python environment. 

## Install DocArray

> To install `DocArray` to follow this user guide, you can use the following command:

Reviewer: ```suggestion
To install `DocArray`, you can use the following command:
```

```console
pip install "docarray[full]"
```

> This will install the main dependencies of `DocArray` and will work will all the supported data modalities.

Reviewer: ```suggestion
This will install the main dependencies of `DocArray` and will work with all the supported data modalities.
```

!!! note 
    To install a very light version of `DocArray` with only the core dependencies, you can use the following command:
    ```
    pip install "docarray"
    ``` 
    

>     If you want to use protobuf and DocArray you can run:

Reviewer: ```suggestion
    If you want to use `protobuf` and `DocArray`, you can run:
```

# User Guide - Intro

This user guide shows you how to use `DocArray` with most of its features.

> They are three main sections:

Author: I don't understand what you mean

- [Representing Data](representing/first_step.md): This section will show you how to use `DocArray` to represent your data.
- [Sending Data](sending/first_step.md): This section will show you how to use `DocArray` to send your data.
- [Storing Data](storing/first_step.md): This section will show you how to use `DocArray` to store your data.

You should first start by reading the [Representing Data](representing/first_step.md) section and both the [Sending Data](sending/first_step.md) and [Storing Data](storing/first_step.md) sections can be read in any order.

> You wil first need to install `DocArray` in you python environment. 

Reviewer 2: ```suggestion
You will first need to install `DocArray` in your Python environment. 
```

## Install DocArray

To install `DocArray` to follow this user guide, you can use the following command:

```console
pip install "docarray[full]"
```

This will install the main dependencies of `DocArray` and will work will all the modalities supported.

!!! note 
    To install a very light version of `DocArray` with only the core dependencies, you can use the following command:
    ```
    pip install "docarray"
    ``` 
    

>     If you want to install user protobuf with minimal dependencies you can do

Reviewer 2: `install user protobuf`? what does that mean?

    ```
    pip install "docarray[common]"
    ``` 

> Depending on your usage you might want to only use `DocArray` with only a couple of specific modalities. 

Reviewer 2: ```suggestion
Depending on your usage you might want to use `DocArray` with only a couple of specific modalities and their dependencies. 
```

> For instance let's say you only want to work with images, you can install `DocArray` using the following command:

Reviewer 2: ```suggestion
For instance, let's say you only want to work with images, you can install `DocArray` using the following command:
```
